src/consumer/runner.js

```diff
@@ -135,20 +150,14 @@ module.exports = class Runner {
 
   waitForConsumer() {
     return new Promise(resolve => {
-      const scheduleWait = () => {
-        this.logger.debug('waiting for consumer to finish...', {
-          groupId: this.consumerGroup.groupId,
-          memberId: this.consumerGroup.memberId,
-        })
-
-        setTimeout(() => (!this.consuming ? resolve() : scheduleWait()), 1000)
-      }
-
       if (!this.consuming) {
         return resolve()
       }
-
-      scheduleWait()
+      this.logger.debug('waiting for consumer to finish...', {
+        groupId: this.consumerGroup.groupId,
+        memberId: this.consumerGroup.memberId,
+      })
+      this.once(CONSUMING_STOP, () => resolve())
```

Comment:
This could be simplied to `this.once(CONSUMING_STOP, resolve)`

Reply:
Yes could be. The reason I use wrapper function is safety. Promise must be resolved with no value, while passing `resolve` as event handler directly may have side effects, promise will be resolved with data passed into event. Ofcourse right now this event has no attached data so expression could be simplified (right now). But what if someone will modify event in future?

Reply:
Fair enough, thanks!